scripts/check_compliance.py

```diff
@@ -165,6 +167,23 @@ def run(self):
         # single Kconfig tree and warns for all references to undefined symbols
         os.environ["KCONFIG_STRICT"] = "y"
 
+        kconfig_env_var_list_path = os.path.join(self.zephyr_base,
+                                    os.path.normpath("cmake/kconfig_env_vars.txt"))
+        # Skipping last new line which gives empty string
+        kconfig_env_var_list = open(kconfig_env_var_list_path).read().split("\n")[:-1]
+
+        # Faster to check for all than iterate over every variable
+        if not all(env_var in os.environ for env_var in kconfig_env_var_list):
```

Comment:
Sets are pretty handy here:

```python
kconfig_env_vars = set(open(...).readlines())

missing_vars = kconfig_env_vars.difference(os.environ)
if missing_vars:
    ...
```

Reply:
That's a great idea I'll do that instead :) Thanks for the tip

Reply:
Sets was much nicer thanks once again 👍